gio-subclass/src/auto/action_group.rs

```diff
@@ -0,0 +1,356 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 49c86ce+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 47c69e6)
+// DO NOT EDIT
+
+#[allow(unused_imports)]
+use libc::{c_int, c_char, c_uchar, c_float, c_uint, c_double,
+    c_short, c_ushort, c_long, c_ulong,
+    c_void, size_t, ssize_t, intptr_t, uintptr_t, time_t, FILE};
+
+#[allow(unused_imports)]
+use glib_ffi::{gboolean, gconstpointer, gpointer, GType};
+
+use gio;
+use gio_ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use libc;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+use gobject_subclass::anyimpl::*;
+use gobject_subclass::object::*;
+
+
+pub trait ActionGroupImpl: AnyImpl + 'static {
+
+    fn action_added(&self, action_group: &gio::ActionGroup, action_name: &str);
+
+    fn action_enabled_changed(&self, action_group: &gio::ActionGroup, action_name: &str, enabled: bool);
+
+    fn action_removed(&self, action_group: &gio::ActionGroup, action_name: &str);
+
+    fn action_state_changed(&self, action_group: &gio::ActionGroup, action_name: &str, state: &glib::Variant);
+
+    fn activate_action(&self, action_group: &gio::ActionGroup, action_name: &str, parameter: Option<&glib::Variant>);
+
+    fn change_action_state(&self, action_group: &gio::ActionGroup, action_name: &str, value: &glib::Variant);
+
+    fn get_action_enabled(&self, action_group: &gio::ActionGroup, action_name: &str) -> bool;
+
+    fn get_action_parameter_type(&self, action_group: &gio::ActionGroup, action_name: &str) -> Option<glib::VariantType>;
```

Comment:
This and `get_action_state_type` have the same problem as in the `Action` interface. The return value must be valid for the lifetime of the object.